src/main/java/org/minions/devfund/bruno/html/HtmlValidator.java

```diff
@@ -0,0 +1,46 @@
+package org.minions.devfund.bruno.html;
+
+import java.util.Queue;
+import java.util.Stack;
+
+/**
+ * Represents the validate the information about a single HTML tags.
+ */
+public final class HtmlValidator {
+
+    /**
+     * Constructor private.
+     */
+    private HtmlValidator() {
+        // Default Constructor.
+    }
+
+    /**
+     * Takes as input a Queue of HtmlTags and return a Stack of HtmlTags that verifies the correctness
+     * of the tag structure.
+     *
+     * @param tags queue tags.
+     * @return a Stack.
+     */
+    public static Stack<HtmlTag> isValidHtml(final Queue<HtmlTag> tags) {
+        Stack<HtmlTag> stack = new Stack<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)